R/github-action.R

```diff
@@ -22,15 +22,18 @@ use_github_action <- function(name,
                               ignore = TRUE,
                               open = FALSE) {
 
-  # Append a `.yaml` extension if needed
-  stopifnot(is_string(name))
+  # Check if a custom URL is being used.
+  if (is.null(url)) {
+    stopifnot(is_string(name))
 
-  if (!grepl("[.]yaml$", name)) {
-    name <- paste0(name, ".yaml")
-  }
+    # Append a `.yaml` extension if needed
+    if (!grepl("[.]yaml$", name)) {
```

Comment:
Just based on reading other comments in this review this makes me wonder if a `name` with pre-existing`.yml` extension will now get an additional `.yaml` extension? Which would be undesirable.

If I'm off-base, just ignore me.

If I'm not, perhaps we should add a test that hits the `.yaml` vs `.yml` thing.

Reply:
@jennybc it will and also cause an error because `r-lib/actions` are labelled as `.yaml`. So, let's patch that!

Reply:
Should be addressed with a `gsub` for `yml` to `yaml` in df603d0 and associated testing infrastructure in 820b76d.